src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,95 @@
+package duke;
+
+import java.util.List;
+
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.InvalidInputException;
+import duke.exceptions.InvalidTaskNumberException;
+import duke.exceptions.MissingTaskNumberException;
+import duke.task.Task;
+
+/**
+ * <code>Parser</code> class deals with making sense of the user command.
+ */
+public class Parser {
+    protected static String task;
+    protected String userInput;
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param userInput User input in the command line.
+     */
+    public Parser(String userInput) {
+        this.userInput = userInput;
+    }
```

Comment:
From an object point of view, might it not be a bit odd to create a new Parser object for every input the user makes?

Reply:
I am inclined to agree with guanyz here. If possible, hope you would consider some of the possible modifications stated here.